wolvic/browser/vr/wvr_manager.cc

```diff
@@ -850,4 +875,62 @@ void WvrManager::UpdateLayerBounds(int16_t frame_index,
   CreateOrResizeWebXrSurface(source_size);
 }
 
+device::mojom::XREnvironmentBlendMode
+WvrManager::PickEnvironmentBlendModeForSession(
+    device::mojom::XRSessionMode session_mode) {
+  DCHECK(wvr_api_->get_system_state().enumerationCompleted);
+
+  const auto& display_state = wvr_api_->get_system_state().displayState;
+  auto supportsBlendMode =
+      [&display_state](const mozilla::gfx::VRDisplayBlendMode blend_mode) {
+        for (auto& supported_blend_mode : display_state.blendModes) {
+          if (supported_blend_mode ==
+              mozilla::gfx::VRDisplayBlendMode::_empty) {
+            return false;
+          }
+          if (supported_blend_mode == blend_mode) {
+            return true;
+          }
+        }
+        return false;
+      };
+
+  switch (session_mode) {
+    case device::mojom::XRSessionMode::kImmersiveVr:
+      if (supportsBlendMode(mozilla::gfx::VRDisplayBlendMode::Opaque)) {
```

Comment:
nit: do we need brackets here ?

Reply:
They're added by git cl format, so I guess yes